### DISCUSSION TITLE
Cannot get non-formatted help output if Rich is installed

### DISCUSSION THREAD

**Discussion by sco1** [fastapi/typer#853] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorials in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer


cli = typer.Typer(rich_markup_mode=None)
print(f"rich markup mode is: {cli.rich_markup_mode}")

@cli.command()
def foo():
    print("hi")

if __name__ == "__main__":
    cli()
```

### Description

As the title says, it doesn't appear possible to get a non-formatted help output if rich is installed, even when `rich_markup_mode` is implicitly or explicitly set to `None`. I'm not quite sure I'm following the flow correctly but I think the issue comes from this logic:

https://github.com/tiangolo/typer/blob/04eba6b70203287176d2823753513226bf778872/typer/core.py#L671-L678

Which is set for `TyperCommand` and `TyperGroup`, and only considers whether or not rich is installed. Uninstalling rich gives me the help output I'm looking for, but I'd like it installed to use elsewhere in my project.

### Operating System

Windows, macOS

### Typer Version

0.12.3

### Python Version

Python 3.12.4

**Comment by svlandeg** [comment, marked as answer]
Hi! Yes, this does seem to be a bug in `typer`. This PR attempts to resolve the issue, and covers the same code path you've mentioned as well: https://github.com/tiangolo/typer/pull/726. The PR is currently work-in-progress but should be in a good state to merge relatively soon.

**Reply by sco1** [reply]
Oops, I didn't think to search PRs, just the issues 🤦 

Thank you for the reference, will keep an eye out!
